plugins/video/pkg_src/src/decoder/video_decoder_mixed.cc

```diff
@@ -101,15 +93,40 @@ void VideoDecoderMixed::Run(dali::Workspace &ws) {
         false /*_enableasyncallocations*/, false);
 
 
-    uint8_t *pVideo = NULL;
-    int nVideoBytes = 0;
-    while (sample.demuxer_->Demux(&pVideo, &nVideoBytes)) {
-      if (nVideoBytes) {
-        auto vecTupFrame = sample.decoder_->Decode(pVideo, nVideoBytes);
+    uint8_t *pVideo = NULL, *pFrame = nullptr;
+    int nVideoBytes = 0, nFrameReturned = 0, nFrame = 0;
+
+    int num_frames = 0;
+
+    do {
+      sample.demuxer_->Demux(&pVideo, &nVideoBytes);
+      nFrameReturned = sample.decoder_->Decode(pVideo, nVideoBytes);
+
+      for (int i = 0; i < nFrameReturned; i++) {
+        pFrame = sample.decoder_->GetFrame();
+
+        uint8_t *dpFrame = output_data + num_frames * sample.demuxer_->GetHeight() *
+                                             sample.demuxer_->GetWidth() * 3;
+        int nWidth = sample.decoder_->GetWidth();
+        int nPitch = sample.decoder_->GetWidth();
+        int iMatrix =
+            sample.decoder_->GetVideoFormatInfo().video_signal_description.matrix_coefficients;
+        bool full_range =
+            sample.decoder_->GetVideoFormatInfo().video_signal_description.video_full_range_flag;
+
+
+        yuv_to_rgb(pFrame, nPitch, reinterpret_cast<uint8_t *>(dpFrame),
+                   sample.decoder_->GetWidth() * 3, sample.decoder_->GetWidth(),
+                   sample.decoder_->GetHeight(), full_range, cuStream);
+        CUDA_CALL(cudaStreamSynchronize(cuStream));
```

Comment:
Why is this synchronization needed here?

Reply:
It was leftover from the more extended version where we might want to do some postprocessing here. Removed it for now, I will add it if needed later.